roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,49 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+from openshift_checks.mixins import NotContainerized
+
+class DockerImageAvailability(NotContainerized, OpenShiftCheck):
+    """Check that required Docker images are available."""
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    def run(self, tmp, task_vars):
+        openshift_release = get_var(task_vars, "openshift_release")
+
+        args = {
+            "name": self.rpm_docker_images(),
+        }
+
+        docker_images = self.module_executor("docker_image_facts", args, tmp, task_vars)
+
```

Comment:
Perhaps the first thing I'd do after calling the module is checking for a "failed" key with value True. If it failed, we can abort immediately.